src/Application.vala

```diff
@@ -21,6 +21,7 @@
 
 namespace Akira {
 	public Akira.Services.Settings settings;
+	public Akira.Services.EventBus event_bus;
```

Comment:
Sadly, we can't have this event bus as a global variable. This will cause issues when we have more than one instance of the app running at the time. Instead, this bus should be a part of the AkiraWindow, and we need to pass the bus object along the way :)

Reply:
Isn't there any way in which we can get the current window from the actual process we are running the window in? 

Simply a way to determine the window we are in from the function call stack or something?
Otherwise we might have to *honor the chain* of references from the AkiraWindow to the component everytime we want it to use the `event_bus` (which might be - you know - quite often). 

So, in order to solve this *issue* do I have to give the caller **and** the receiver a reference to the current window and create the `event_bus` as a local variable to that window, right?

Reply:
Mmmm as far as I know, there isn't a real way to do it.... The whole app is under the same process.... But we could probably do it by adding a function to get the newest window from the application, and have everything that uses the state hold a reference to it. That way when a new window is created, the old widgets still hold a reference to the original window

Reply:
OK I just changed from global variable to local for each window. 

The downside of this approach is that you have to hold a reference to the *current* window in every widget/component that needs access to `event_bus`. 
This might be annoying since we need to actually provide this reference from parent to children, for every nested object.
 
Maybe a different way to solve this problem is to have the main application to be a global variable and have methods on it to retrieve the *last* created window, so during UI construction's process we can attach the reference to window to the consumer widget with a function (something like `application.get_last_window ()`) instead than having this declared in the object constructor. 

I'll maybe try to code a prototype to show off, but in the meantime I did what you asked :)